hoodie-client/src/main/java/com/uber/hoodie/io/compact/CompactionOperation.java

```diff
@@ -33,10 +35,10 @@
  */
 public class CompactionOperation implements Serializable {
 
-  private String dataFileCommitTime;
-  private long dataFileSize;
+  private Optional<String> dataFileCommitTime;
```

Comment:
Can you check if you can properly serialize/deserialize this object. AFAIK, Optional is not serializable

Reply:
I think kryo includes some serializers.?

Reply:
@vinothchandar @n3nash : Are we not going to support plain JavaSerializer as an option ?

Reply:
Most folks use kryo. for now, only plan on supporting kryo since things like avro serialization don't work otherwise.. I think its okay to assume this unless someone really makes a case for java ser